pyretailscience/segmentation/threshold.py

```diff
@@ -0,0 +1,105 @@
+"""Threshold-Based Customer Segmentation Module.
+
+This module provides the `ThresholdSegmentation` class, which segments customers
+based on user-defined thresholds and segment mappings.
+
+Key Features:
+- Segments customers based on specified percentile thresholds.
+- Uses a specified column for segmentation, with an aggregation function applied.
+- Handles customers with zero spend using configurable options.
+- Utilizes Ibis for efficient query execution.
+"""
+
+from typing import Literal
+
+import ibis
+import pandas as pd
+
+from pyretailscience.options import get_option
+from pyretailscience.segmentation.base import BaseSegmentation
+
+
+class ThresholdSegmentation(BaseSegmentation):
+    """Segments customers based on user-defined thresholds and segments."""
+
+    _df: pd.DataFrame | None = None
+
+    def __init__(
+        self,
+        df: pd.DataFrame | ibis.Table,
+        thresholds: list[float],
+        segments: dict[any, str],
+        value_col: str | None = None,
+        agg_func: str = "sum",
+        zero_segment_name: str = "Zero",
+        zero_value_customers: Literal["separate_segment", "exclude", "include_with_light"] = "separate_segment",
+    ) -> None:
+        """Segments customers based on user-defined thresholds and segments.
+
+        Args:
+            df (pd.DataFrame | ibis.Table): A dataframe with the transaction data. The dataframe must contain a customer_id column.
+            thresholds (List[float]): The percentile thresholds for segmentation.
+            segments (Dict[str, str]): A dictionary where keys are segment IDs and values are segment names.
```

Comment:
Similar issue here with the type being the old type
```suggestion
            segments (List[str]): A list of segment names for each threshold.
```